stream_app.py

```diff
@@ -0,0 +1,43 @@
+import os
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, from_json
+from pyspark.sql.types import StructType, StructField, DoubleType
+
+# Initialize spark env
+# spark = SparkSession.builder.master('local').getOrCreate()
+spark = SparkSession.builder.appName('StructredStreamingApp').getOrCreate()
+
+# Azure setting information
+EVENTHUB_CONNECTION_STRING = os.getenv('EVENTHUB_CONNECTION_STRING')
+EVENTHUB_NAMESPACE = os.getenv('EVENTHUB_NAMESPACE')
+EVENT_HUB_NAME = os.getenv('EVENT_HUB_NAME') # same as kafka topic
+
+# Kafka stream settings
+EH_SASL = f'org.apache.kafka.common.security.plain.PlainLoginModule required \
+    username="$ConnectionString" password="{EVENTHUB_CONNECTION_STRING}";'
+BOOTSTRAP_SERVERS = f'{EVENTHUB_NAMESPACE}:9093'
+
+rowStreamDf = spark.readStream \
+    .format("kafka") \
+    .option("kafka.sasl.mechanism", "PLAIN") \
+    .option("kafka.security.protocol", "SASL_SSL") \
+    .option("kafka.sasl.jaas.config", EH_SASL) \
+    .option("kafka.bootstrap.servers", BOOTSTRAP_SERVERS) \
+    .option("subscribe", EVENT_HUB_NAME) \
+    .load() \
+    .select(col("value").cast("STRING"))
```

Comment:
Connect IoT Hub. IoT Hub has Event Hub compatible build-in endpoint. Event Hub is compatible with Kafka.
In some project with customer, they want to utilize Kafka connector because they can switch to use Kafka in the  on premises environment, so I uses Kafka connector in this repo.

Reply:
1.	No error handling?
2.	Instead of adding a lot of options directly into readstream, you can also use MAP like below. It might be better for error handling, testability and readability. 

I don’t have a python sample but scalar. 

val eventhubParameters = Map[String, String](
  "eventhubs.policyname" -> policyName,
  "eventhubs.policykey" -> policyKey,
  "eventhubs.namespace" -> eventHubNamespace,
  "eventhubs.name" -> eventHubName,
  "eventhubs.partition.count" -> partitionCount,
  "eventhubs.consumergroup" -> consumerGroup,
  "eventhubs.progressTrackingDir" -> progressDir
)

val inputStream = spark.readStream.
  format("eventhubs").
  options(eventhubParameters).
  load

Reply:
Thank you for your feedback. I updated code.